sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateUnsafeRowJoinerSuite.scala

```diff
@@ -99,6 +101,23 @@ class GenerateUnsafeRowJoinerSuite extends SparkFunSuite {
     testConcatOnce(N, N, variable)
   }
 
+  test("SPARK-30993: UserDefinedType matched to fixed length SQL type shouldn't be corrupted") {
```

Comment:
BTW, @HeartSaVioR . How do you want to handle SPARK-30986 after SPARK-30993?

Reply:
Actually I missed the existence of SPARK-30986 and created SPARK-30993. I left my issue as it is, because SPARK-30986 is describing only one of issue SPARK-30993 could incur - SPARK-30993 has a broader coverage.

I can maybe add E2E test for SPARK-30986 here (though not sure we want to use two JIRA issues here), but wondering is it preferable to add SS test if it can be reproduced in batch query test. From my side it's easier to add SS test, because SPARK-30986 has it - just need to refine a bit.

Reply:
Please let me know if we would like to use SPARK-30986 instead - I can simply mark SPARK-30993 as duplicated and change the PR title as well as the code. Thanks!